sources/packages/web/src/components/aest/institution/modals/DeclineProgramModal.vue

```diff
@@ -1,71 +1,89 @@
+    <modal-dialog-base
+      title="Decline program"
+      :showDialog="showDialog"
+      max-width="730"
+    >
+      <template #content>
+        <error-summary :errors="declineProgramForm.errors" />
+        <div class="pb-2">
+          <span class="label-value"
+            >Outline the reasoning for declining this program. This will be
+            stored in the institution profile notes.</span
+          >
+        </div>
+        <div class="pb-2">
+          <span class="label-bold">Notes</span>
+        </div>
+        <v-textarea
+          label="Long text..."
+          v-model="formModel.declinedNote"
+          variant="outlined"
+          :rules="[(v) => !!v || 'Notes is required']"
+      /></template>
+      <template #footer>
+        <check-permission-role :role="Role.InstitutionApproveDeclineProgram">
+          <template #="{ notAllowed }">
+            <footer-buttons
+              :processing="processing"
+              primaryLabel="Decline now"
+              @primaryClick="submit"
+              @secondaryClick="cancel"
+              :disablePrimaryButton="notAllowed"
+            />
+          </template>
+        </check-permission-role>
+      </template>
+    </modal-dialog-base>
+  </v-form>
 </template>
 
 <script lang="ts">
+import { ref, reactive } from "vue";
 import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import ErrorSummary from "@/components/generic/ErrorSummary.vue";
 import { useModalDialog } from "@/composables";
 import { DeclineProgramAPIInDTO } from "@/services/http/dto";
 import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
-import { Role } from "@/types";
+import { Role, VForm } from "@/types";
 
 export default {
   components: {
     ModalDialogBase,
     CheckPermissionRole,
+    ErrorSummary,
   },
   setup() {
-    const { showDialog, resolvePromise, showModal } = useModalDialog<
-      DeclineProgramAPIInDTO | undefined
+    const { showDialog, showModal, resolvePromise } = useModalDialog<
+      DeclineProgramAPIInDTO | false
     >();
-    let declineProgramForm: any = undefined;
+    const declineProgramForm = ref({} as VForm);
+    const formModel = reactive({
+      declinedNote: "",
+    } as DeclineProgramAPIInDTO);
 
-    const declineProgram = () => {
-      return declineProgramForm.submit();
-    };
-    const submitForm = (formData: DeclineProgramAPIInDTO) => {
-      resolvePromise(formData);
-    };
-
-    const formLoaded = (form: any) => {
-      declineProgramForm = form;
+    const submit = async () => {
+      const validationResult = await declineProgramForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      resolvePromise(formModel);
     };
 
-    const dialogClosed = () => {
-      resolvePromise(undefined);
+    const cancel = () => {
+      declineProgramForm.value.reset();
+      resolvePromise(false);
+      declineProgramForm.value.errors = [];
```

Comment:
The form has a method to reset the validation, is it not working?

Reply:
yes there is `resetValidation()`. I also did this way in my previous PR. I will fix it in my next PR
![image](https://user-images.githubusercontent.com/77353155/191114833-bd58ea15-55ee-4cd1-a86f-c3dbce8e8b05.png)